engine/src/test/java/org/camunda/bpm/engine/test/api/repository/ProcessDefinitionQueryTest.java

```diff
@@ -697,6 +730,22 @@ public void testQueryByNoVersionTag() {
       .count()).isEqualTo(4);
   }
 
+  @Test
+  public void testQueryOrderByDeployTime() {
```

Comment:
I suggest to explictly assert the order in which process definitions are supposed to be returned. By these kind of dynamic assertions you always run into the risk of loops being skipped entirely or partially, if the results of the APIs change unintendedly (e.g. the test is green if no definitions are returned at all). Check for example `ExecutionQueryTest#testQuerySorting` for a more robust test.